File: wallet/src/main/java/com/worldventures/wallet/analytics/BaseSetDefaultCardAction.java
package com.worldventures.wallet.analytics;


import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.wallet.domain.entity.record.Record;

abstract class BaseSetDefaultCardAction extends BaseCardDetailsWithDefaultAction {

   @Attribute("setdefaultcard") final String setDefaultCard = "1";
   @Attribute("setdefaultwhere") String setDefaultWhere;

   @Override
   public void fillRecordDetails(Record record) {
      super.fillRecordDetails(record);
      defaultPaycard = "Yes";
   }
}

File: wallet/src/main/java/com/worldventures/wallet/analytics/BaseSetDefaultCardAction.kt
package com.worldventures.wallet.analytics

import com.worldventures.core.service.analytics.Attribute
import com.worldventures.wallet.domain.entity.record.Record

abstract class BaseSetDefaultCardAction : BaseCardDetailsWithDefaultAction() {

   @Attribute("setdefaultcard")
   val setDefaultCard = "1"
   @Attribute("setdefaultwhere")
   var setDefaultWhere: String? = null

   override fun fillRecordDetails(record: Record) {
      super.fillRecordDetails(record)
      defaultPaycard = "Yes"
   }
}
